samcli/local/lambdafn/config.py

```diff
@@ -78,3 +78,6 @@ def __init__(
         self.env_vars.handler = self.handler
         self.env_vars.memory = self.memory
         self.env_vars.timeout = self.timeout
+
+    def __eq__(self, other):
+        return self.name == other.name
```

Comment:
Should this only be name to consider the objects equal?

Reply:
I think per template, we can not have 2 lambda functions that have the same name, that is why I think name is good for comparison.